jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/RetainableByteBuffer.java

```diff
@@ -159,6 +170,419 @@ default void clear()
         BufferUtil.clear(getByteBuffer());
     }
 
+    default int append(byte[] bytes, int offset, int length)
+    {
+        ByteBuffer to = getByteBuffer();
+
+        int pos = BufferUtil.flipToFill(to);
+        try
+        {
+            length = Math.min(length, to.remaining());
+            to.put(bytes, offset, length);
+        }
+        finally
+        {
+            BufferUtil.flipToFlush(to, pos);
+        }
+        return length;
+    }
+
+    default void append(ByteBuffer bytes)
+    {
+        BufferUtil.append(getByteBuffer(), bytes);
+    }
+
+    default boolean append(ByteBuffer bytes, Runnable releaser)
+    {
+        append(bytes);
+        if (bytes.hasRemaining())
+            return false;
+        releaser.run();
+        return true;
+    }
+
+    default void append(RetainableByteBuffer bytes)
+    {
+        bytes.writeTo(this);
+    }
+
+    default void putTo(ByteBuffer toInfillMode)
+    {
+        ByteBuffer slice = getByteBuffer().slice();
+        toInfillMode.put(slice);
+    }
+
+    default boolean writeTo(ByteBuffer to)
+    {
+        ByteBuffer slice = getByteBuffer().slice();
+        BufferUtil.append(to, slice);
+        return slice.remaining() == 0;
+    }
+
+    default boolean writeTo(RetainableByteBuffer to)
+    {
+        if (remaining() == 0)
+            return true;
+        ByteBuffer slice = getByteBuffer().slice();
+        to.append(slice);
+        return slice.remaining() == 0;
+    }
+
+    default void writeTo(OutputStream out) throws IOException
+    {
+        BufferUtil.writeTo(getByteBuffer(), out);
+    }
+
+    default void writeTo(Content.Sink sink, boolean last, Callback callback)
+    {
+        sink.write(last, getByteBuffer(), callback);
+    }
+
+    static RetainableByteBuffer newAggregator(ByteBufferPool pool, boolean direct, int growBy, int maxCapacity)
```

Comment:
What is this factory method buying us over directly instantiating the `Aggregator` class?